beacon_node/network/src/network_beacon_processor/gossip_methods.rs

```diff
@@ -440,11 +479,46 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
         packages: Vec<GossipAggregatePackage<T::EthSpec>>,
         reprocess_tx: Option<mpsc::Sender<ReprocessQueueMessage>>,
     ) {
-        let aggregates = packages.iter().map(|package| package.aggregate.as_ref());
+        let (to_process, skip): (
+            Vec<(
+                GossipAggregatePackage<T::EthSpec>,
+                SignedAggregateAndProof<T::EthSpec>,
+            )>,
+            Vec<(AttnError, GossipAggregatePackage<T::EthSpec>)>,
+        ) = packages.into_iter().partition_map(|p| {
+            match self
+                .chain
+                .is_lazy_att_observed_subset(&p.aggregate.message.aggregate)
+            {
+                Ok(false) => match p.aggregate.clone().not_lazy() {
+                    Ok(agg) => itertools::Either::Left((p, agg)),
+                    Err(e) => itertools::Either::Right((e.into(), p)),
+                },
+                Ok(true) => itertools::Either::Right((
+                    AttnError::AttestationSupersetKnown(
+                        p.aggregate.message.aggregate.data.tree_hash_root(),
+                    ),
+                    p,
+                )),
+                // The only error that could occur here is observed_attestations::Error::SlotTooLow
+                Err(beacon_chain::ObservedAggregatesError::SlotTooLow {
+                    slot,
+                    lowest_permissible_slot,
+                }) => itertools::Either::Right((
+                    AttnError::PastSlot {
+                        attestation_slot: slot,
+                        earliest_permissible_slot: lowest_permissible_slot,
+                    },
+                    p,
+                )),
+                // This should never occur
+                Err(_) => itertools::Either::Right((AttnError::InvalidSignature, p)),
+            }
+        });
```

Comment:
I added a log here just for fun, although I think the metrics will probably show us the increase in aggregates skipped

```rust
debug!(
    self.log,
    "Processing aggregate batch";
    "skipped" => skip.len(),
    "total" => skip.len() + to_process.len(),
);
```